lib/core/sync_job_runner.rb

```diff
@@ -59,7 +59,10 @@ def do_sync!
 
         Utility::Logger.debug("#{existing_ids.size} documents are present in index #{@connector_settings.index_name}.")
 
-        @connector_instance.yield_documents do |document|
+        filtering = @connector_settings.filtering
+        job_description = { :filtering => filtering }
```

Comment:
Why not just pass filtering rules as is?

Reply:
Will be removed with the next PR update.

Reply:
Actually this will probably be removed, when we change the return value by `claim_job` (when we return the full document, we can probably just pass it as is; added a comment to hint at this future change).